src/components/flash/flash.js

```diff
@@ -133,9 +135,14 @@ class Flash extends React.Component {
    * @method componentWillReceiveProps
    * @return(Void)
    */
-  componentWillReceiveProps(prevProps) {
-    if (prevProps.open !== this.props.open) {
-      this.setState({ dialogs: {} });
+  componentWillReceiveProps(nextProps) {
+    clearTimeout(this.removePortalTimeout);
+    if (nextProps.open) {
+      this.setState({ dialogs: {}, open: nextProps.open });
+    } else {
```

Comment:
this should have some additional guard on it so it doesn't call it unnecessarily.